cpp/include/cudf/types.hpp

```diff
@@ -17,9 +17,23 @@
 #pragma once
 
 #ifdef __CUDACC__
+/**
+ * @brief Indicates that the function or method is usable on host and device
+ */
 #define CUDF_HOST_DEVICE __host__ __device__
+/**
+ * @brief Indicates that the function is a CUDA kernel
+ */
+#define CUDF_KERNEL __global__ static
```

Comment:
note (non-blocking): I don't think anyone builds libcudf with `rdc=true`, but if you wanted to be extra pedantic, then `CUDF_KERNEL` should expand to `__attribute__ ((visibility ("hidden")))` when `__CUDACC_RDC__` is defined in order to preserve the binary size improvements that come from symbol deduplication within the DLL with `rdc=true`.